packages/lambdapi/lambdapi.2.1.0/opam

```diff
@@ -0,0 +1,82 @@
+opam-version: "2.0"
+synopsis: "Proof assistant for the λΠ-calculus modulo rewriting"
+description: """
+This package provides:
+- A lambdapi command for checking .lp or .dk files,
+translating .dk files to .lp files and vice versa,
+or launching an LSP server for editing .lp files.
+- A library of logic definitions and basic definitions and proofs
+on natural numbers and polymorphic lists.
+- A rich Emacs mode based on LSP (available on MELPA too).
+- A basic mode for Vim.
+- OCaml libraries.
+A VSCode extension is available on the VSCode Marketplace.
+
+Find Lambdapi user manual on https://lambdapi.readthedocs.io/.
+
+Lambdapi provides a rich type system with dependent types.
+In Lambdapi, one can define both type and function symbols
+by using rewriting rules (oriented equations). The declaration
+of symbols and rewriting rules is separated so that one can
+easily define inductive-recursive types for instance.
+Rewrite rules can be exported to the TRS and XTC formats
+for checking confluence and termination with external tools.
+A symbol can be declared associative and commutative.
+Lambdapi supports unicode symbols and infix operators.
+
+Lambdapi does not come with a pre-defined logic. It is a
+powerful logical framework in which one can easily define
+its own logic and build and check proofs in this logic.
+There exist .lp files defining first or higher-order logic
+and complex type systems like in Coq or Agda.
+
+Lambdapi provides a basic module and package system,
+interactive modes for proving both unification goals
+and typing goals, and tactics for solving them step by step.
+In particular, a rewrite tactic like in SSReflect, and
+a why3 tactic for calling external automated provers through
+the Why3 platform."""
+maintainer: ["dedukti-dev@inria.fr"]
+authors: ["Deducteam"]
+license: "CECILL-2.1"
+homepage: "https://github.com/Deducteam/lambdapi"
+bug-reports: "https://github.com/Deducteam/lambdapi/issues"
+depends: [
+  "dune" {>= "2.7"}
+  "ocaml" {>= "4.08.0"}
+  "menhir" {>= "20200624" & < "20211223"}
```

Comment:
```suggestion
  "menhir" {>= "20200624"}
```
Is there a reason why this constraint is still there? I see you are using `(using menhir 2.0)` so it should be compatible with 20211223

Reply:
I am not sure that why3 is compatible with the last version of menhir.
And why3 is not compatible with the last version of alt-ergo (used in some lambdapi tests).
See https://gitlab.inria.fr/why3/why3/-/issues/601.
and https://github.com/OCamlPro/alt-ergo/issues/492. It seems to have been fixed on the master branch of alt-ergo (https://github.com/OCamlPro/alt-ergo/pull/495) but not released yet.

Reply:
The latest version of menhir came with some breaking changes that required us to add lots of upper bounds. All versions of alt-ergo currently have the upper bound, so for the moment you would get it implicitly.

My suggestion is to remove it from here, and once alt-ergo and why3 are updated, if we see that it breaks lambdapi, we will re-add it and inform you. What do you think?